koku/masu/api/report_data.py

```diff
@@ -118,14 +164,37 @@ def report_data(request):
                 ).apply_async(queue=queue_name or fallback_queue)
                 async_results.append({str(month): str(async_result)})
         else:
-            # TODO: when DEVELOPMENT=False, disable resummarization for all providers to prevent burning the db.
-            # this query could be re-enabled if we need it, but we should consider limiting its use to a schema.
-            if not settings.DEVELOPMENT:
-                errmsg = "?provider_uuid=* is invalid query."
+            provider_list = [
+                Provider.PROVIDER_AWS,
+                Provider.PROVIDER_GCP,
+                Provider.PROVIDER_AZURE,
+                Provider.PROVIDER_OCI,
+                Provider.PROVIDER_OCP,
+            ]
+            if not provider_type:
+                errmsg = "provider_type is required when resummarizing all"
                 return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
-            for month in months:
-                async_result = update_all_summary_tables.delay(month[0], month[1], invoice_month=month[2])
-                async_results.append({str(month): str(async_result)})
+            if provider_type not in provider_list:
+                errmsg = f"unrecongized provider_type: {provider_type}"
+                return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
+            key_exist, timeout = get_cached_resummarize_by_provider_type(provider_type)
+            if key_exist:
+                errmsg = f"provider_type ({provider_type}) still disabled for {round(timeout/60, 2)} minutes"
+                return Response({"Error": errmsg}, status=status.HTTP_400_BAD_REQUEST)
+
+            key_set = set_cached_resummarize_by_provider_type(provider_type)
+
+            if key_set:
+                if provider_type == Provider.PROVIDER_OCP:
+                    if ocp_on_cloud_type := params.get("ocp_on_cloud_type"):
```

Comment:
Not sure I understand this line, is this different from >> `ocp_on_cloud = params.get("ocp_on_cloud", "true").lower()`

Reply:
So, the intent was to have `ocp_on_cloud=AWS`, that way you could just resummarize all OCP sources that are connected to an AWS source. However, this was really a last minute change. So, I think we should back it out.